lib/elixir/lib/kernel/special_forms.ex

```diff
@@ -1879,4 +1879,57 @@ defmodule Kernel.SpecialForms do
   special macro. For more information, check the docs for `case/2`.
   """
   defmacro receive(args), do: error!([args])
+
+  @doc """
+  Evaluates to `not(left in right)`.
+
+  See `Kernel.in/2` and `Kernel.not/1` for more information.
+
+  ## Examples
+
+      iex> :missing_value not in []
+      true
+
+      iex> :present_value not in [:present_value]
+      false
+
+  ## Special AST Considerations
+
+  This is rewritten in the Elixir compiler, so `:left not in :right` appears
+  the same as `not(:left in :right)` does in the AST:
+
+      quote do: :left not in :right
+      #=> {:not, [...], [{:in, [...], [:left, :right]}]}
+
+      quote do: not(:left in :right)
+      #=> {:not, [...], [{:in, [...], [:left, :right]}]}
+
+  Additionally, `Macro.to_string/2` will translate all occurrences of
+  `not(:left in :right)` to `:left not in :right`.
+
+      Macro.to_string quote do: not(:left in :right)
+      #=> ":left not in :right"
+
+  ## Guards
+
+  The `not in/2` operator can be used in guard clauses as long as the
+  right-hand side is a range or a list. In such cases, Elixir will expand the
+  operator to a valid guard expression. For example:
+
+      when x not in [1, 2, 3]
+
+  translates to:
+
+      when not(x === 1 or x === 2 or x === 3)
+
+  When using ranges:
+
+      when x not in 1..3
+
+  translates to:
+
+      when not(x >= 1 and x <= 3)
+
+  """
+  defmacro unquote(:"not in")(left, right), do: error!([left, right])
```

Comment:
I wouldn't like to introduce this as a special form because it never appears in the AST (in contrast to the other ones that do appear in the AST). We can move this documentation to `Kernel.in/2`.